activerecord/lib/active_record/attribute_assignment.rb

```diff
@@ -4,21 +4,29 @@ module ActiveRecord
   module AttributeAssignment
     private
       def _assign_attributes(attributes)
-        multi_parameter_attributes = nil
+        multi_parameter_attributes = nested_parameter_attributes = nil
 
         attributes.each do |k, v|
           key = k.to_s
 
           if key.include?("(")
             (multi_parameter_attributes ||= {})[key] = v
+          elsif v.is_a?(Hash)
+            (nested_parameter_attributes ||= {})[key] = v
           else
             _assign_attribute(key, v)
           end
         end
 
+        assign_nested_parameter_attributes(nested_parameter_attributes) if nested_parameter_attributes
         assign_multiparameter_attributes(multi_parameter_attributes) if multi_parameter_attributes
```

Comment:
Good point!

Reply:
I'm not convinced we should be changing anything without a failing test showing why its necessary to change. The previous order was there for 4 years and I haven't seen any issues reported with it.

This PR exists because the Active Record test suite was lacking coverage. What if changing the order is also not covered but some app breaks by changing it?

Reply:
Ok, reverting back the original commit of just adding the test.

Reply:
> If the order as mentioned above should change and needs a test to drive it, it probably belongs in a different PR with a different issue to drive the change. Keeping this scoped to just the bug fix that the original code removal changed is probably most appropriate.

Exactly :+1:

> I could add a test to drive the change in this PR if needed, or I could change this to deal with multiparameter foreign key in a different PR. To me, it makes little difference, I only want to get whatever can get this issue fixed the quickest. My main focus is not having the issue popping up in our local app.

Let's get this issue fixed first and any followup can happen separately